pkg/apis/logging/v1/elasticsearch_types.go

```diff
@@ -185,6 +185,7 @@ type ElasticsearchNodeSpec struct {
 type ElasticsearchStorageSpec struct {
 	// The name of the storage class to use with creating the node's PVC.
 	// More info: https://kubernetes.io/docs/concepts/storage/storage-classes/
+	// +optional
```

Comment:
does this need to be optional since its a pointer to a string? if its omitted it would just be `nil`

Reply:
The `+optional` annotation is the correct way to instruct kubebuilder to omit adding this field to `requiredFields` for OpenAPI schema validation. Prior to this annotation kubebuilder based its decicion on the `omitempty` struct tag, which is not always what you want. In this case both annotation and struct tag apply, because we don't want to change this field to a pointer for backward compatibility. However we want to get empty sentintel value (aka the empty struct) and act upon it.